source/test/main.test.ts

```diff
@@ -1,74 +1,284 @@
 import {
-    App
+  App
 } from 'aws-cdk-lib';
 
-import { Template } from "aws-cdk-lib/assertions";
+import {
+  Template,
+  Match,
+  Capture
+} from "aws-cdk-lib/assertions";
 
 import {
-    SolutionStack
+  SolutionStack
 } from '../src/stack';
 
 import {
-    MainStack
+  MainStack
 } from '../src/molecular-unfolding/cdk/stack-main';
 
 test('can create base stack', () => {
-    const app = new App();
-    const stack = new SolutionStack(app, 'test');
-    Template.fromStack(stack);
+  const app = new App();
+  const stack = new SolutionStack(app, 'test');
+  Template.fromStack(stack);
 });
 
 test('can create MainStack', () => {
-    const app = new App();
-    const stack = new MainStack(app, 'test');
-    expect(stack).not.toBeNull()
+  const app = new App();
+  const stack = new MainStack(app, 'test');
+  expect(stack).not.toBeNull()
 });
 
 test("synthesizes the way we expect", () => {
-    const app = new App();
-    const stack = new MainStack(app, 'test');
-    const template = Template.fromStack(stack);
-    expect(template).toBeTruthy()
+  const app = new App();
+  const stack = new MainStack(app, 'test');
+  const template = Template.fromStack(stack);
+  expect(template).toBeTruthy()
 })
 
 test("has 1 s3 bucket", () => {
```

Comment:
still feel those existing test cases need polish, let's cover it by #25

Reply:
will enhance these test cases.